naga/src/front/wgsl/error.rs

```diff
@@ -738,44 +747,44 @@ impl<'a> Error<'a> {
                 )],
                 notes: vec![],
             },
-            Error::AutoConversion { dest_span, ref dest_type, source_span, ref source_type } => ParseError {
```

Comment:
nitpick: It's preferable to keep destructuring these errors types. That way, we can force ourselves to use the diagnostic information we have, if we ever add any (which happens often).

Reply:
Rust doesn't have matching on `Box`, so the result looks a bit clunky.